src/operations/StandardOperation.cpp

```diff
@@ -347,68 +385,14 @@ namespace qc {
         for (const auto& c: controls) {
             of << " " << qreg[c.qubit].second << ",";
         }
-        for (const auto& target: targets) {
-            of << " " << qreg[target].second << ";\n";
-        }
-        // apply X operations to negate the respective controls again
-        for (const auto& c: controls) {
-            if (c.type == Control::Type::Neg) {
-                of << "x " << qreg[c.qubit].second << ";\n";
-            }
-        }
-    }
-
-    void StandardOperation::dumpOpenQASMSwap(std::ostream& of, const RegisterNames& qreg) const {
-        for (const auto& c: controls) {
-            if (c.type == Control::Type::Neg) {
-                of << "x " << qreg[c.qubit].second << ";\n";
-            }
-        }
-
-        of << std::string(controls.size(), 'c') << "swap";
-        for (const auto& c: controls) {
-            of << " " << qreg[c.qubit].second << ",";
-        }
-        of << " " << qreg[targets[0]].second << ", " << qreg[targets[1]].second << ";\n";
-
-        for (const auto& c: controls) {
-            if (c.type == Control::Type::Neg) {
-                of << "x " << qreg[c.qubit].second << ";\n";
+        for (std::size_t i = 0; i < targets.size(); ++i) {
+            of << " " << qreg[targets[i]].second;
+            if (i < targets.size() - 1) {
+                of << ",";
             }
```

Comment:
Do we have a guarantee that there will be at least one element? For this concatenation I'd find it cleaner to write the first element before the loop and then prepren `","` before each element in the loop. (Feel free to just resolve this)

Reply:
Unfortunately, we don't have such a guarantee (the global phase gate has no targets). However, I think I found a neat solution in https://github.com/cda-tum/qfr/pull/280/commits/be655096613d6138c6b17a0dec430c71f7cae26d.